tests/test_cookiecutter_nested_templates.py

```diff
@@ -0,0 +1,14 @@
+"""Test cookiecutter invocation with nested configuration structure."""
+import os
+
+from cookieninja import main
+
+
+def test_cookiecutter_nested_templates(mocker):
+    """Verify cookiecutter nested configuration files mechanism."""
+    mock_generate_files = mocker.patch("cookieninja.main.generate_files")
+    main_dir = os.path.join("tests", "fake-nested-templates")
```

Comment:
By convention, when invoking functions from modules, we should import the function's parent module.
In this case the function call should be `path.join` instead of `os.path.join`.
This means that you should import the `path` module from the `os` module instead of the `os` module itself.